### PR TITLE
Add Kubernetes Auth Method docs to Vault page.

### PR DESCRIPTION
**Pull request by adityadani**

### CODE REVIEW

File: content/key-management/vault/_index.md [hunk 1]
```diff
@@ -85,6 +104,112 @@ data:
...
+{{</info>}}
+
+You can skip the **Kubernetes Auth Method** section and go to [Step 2](#step-2-setup-vault-as-the-secrets-provider-for-portworx)
+
+#### Using Kubernetes Auth Method
+This method allows Portworx to authenticate with Vault using a Kubernetes service account token. More information on how to setup Kubernetes Vault Auth can be found [here](https://www.vaultproject.io/docs/auth/kubernetes) in the official Vault documentation.
+
+##### Step 1a: Create a ServiceAccount for vault auth delegation
+
+Run the following command to create a ServiceAccount and ClusterRoleBinding. This ServiceAccount grants Portworx permissions to the [Kubernetes TokenReview API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#tokenreview-v1-authentication-k8s-io)
```
Comment (saheienko) [inline content/key-management/vault/_index.md:118]: This service account token is used by vault.

File: content/key-management/vault/_index.md [hunk 2]
```diff
@@ -57,10 +57,30 @@ Portworx requires the following Vault credentials to use its APIs
 
     Name to use as the SNI host when connecting via TLS.
 
+- **Vault Auth Method [VAULT_AUTH_METHOD]**
+
+    Specifies the Auth method that Portworx should use while authenticating with Vault. This field is set only when using Kubernetes Auth Method.
```
Comment (saheienko) [inline content/key-management/vault/_index.md:62]: This parameter will be used for other vault auth methods too. 
Would it be better to say - Available auth methods: kubernetes?

File: content/key-management/vault/_index.md [hunk 3]
```diff
@@ -135,7 +260,7 @@ roleRef:
 EOF
 ```
 
-##### Step 2b: Edit the Portworx DaemonSet
```
Comment (saheienko) [inline content/key-management/vault/_index.md:138]: Is it a required change? (DaemonSet is a valid name)

File: content/key-management/vault/_index.md (hunk 2 above)
Comment (adityadani) [reply content/key-management/vault/_index.md:62]: Makes sense. Will change this.

File: content/key-management/vault/_index.md [hunk 4]
```diff
@@ -85,6 +103,96 @@ data:
...
+
+#### Using Kubernetes Auth Method
+This method allows Portworx to authenticate with Vault using a Kubernetes service account token. More information on how to setup Kubernetes Vault Auth can be found [here](https://www.vaultproject.io/docs/auth/kubernetes) in the official Vault documentation.
+
+##### Step 1a: Create a ServiceAccount for vault auth delegation
+
+Run the following command to create a ServiceAccount and ClusterRoleBinding. This ServiceAccount and its associated token is used by Vault to authenticate requests from Portworx. Vault uses the [Kubernetes TokenReview API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#tokenreview-v1-authentication-k8s-io).
+
+```text
+kubectl create clusterrolebinding vault-tokenreview-binding --clusterrole=system:auth-delegator --serviceaccount=kube-system:vault-auth
```
Comment (saheienko) [inline content/key-management/vault/_index.md:120]: Command for serviceaccount missing:
```
kubectl create serviceaccount vault-auth -n kube-system
```

Comment (adityadani) [reply content/key-management/vault/_index.md:121]: done

**Pull request merged**